lib/cobra_commander/calculated_component_tree.rb

```diff
@@ -121,25 +122,19 @@ def combined_deps(json)
 
       def build_workspaces(workspaces)
         return [] if workspaces.nil?
-        workspaces.map do |workspace|
-          glob = "#{@root_path}/#{workspace}/package.json"
-          workspace_dependencies = Dir.glob(glob)
-          workspace_dependencies.map do |wd|
-            { name: component_name(wd), path: component_path(wd) }
-          end
-        end.flatten
-      end
-
-    private
 
-      def component_name(dir)
-        component_path(dir).split("/")[-1]
+        yarn_workspaces = read_workspaces
+        yarn_workspaces.map do |_, component|
+          { name: component["location"].split("/")[-1], path: component["location"] }
+        end
       end
 
-      def component_path(dir)
-        return dir.split("/package.json")[0] if @root_path == "."
+    private
 
-        dir.split(@root_path)[-1].split("/package.json")[0]
+      def read_workspaces
+        output, = Open3.capture2("yarn workspaces info", chdir: @root_path)
+        striped_output = output.lines[1..-2].join
```

Comment:
If you do `yarn workspaces info --silent`, you don't need this cleanup.

Reply:
good finding. Thanks